code/03_census_and_bed_days.R

```diff
@@ -0,0 +1,868 @@
+  group_by(Healthboard,age_grp,reas1) %>% 
+  summarise(OBDs=sum(OBDs,na.rm=TRUE)) %>% 
+  ungroup()
+
+
+#write_sav(Scotbeddaysdiffage_grp,paste0(filepath,"Scot bed days diff age_grp_R.sav")) # save out file
+
+
+# add files
+
+Scotlandhbbeddays<-rbind(Scotbeddaysdiffage_grp,hbbeddaysdiffage_grp)
+
+# Get Scotland and HB bed days for all ages
+
+datafile13<-Scotlandhbbeddays %>% mutate(age_grp="All")
+
+Scotlandhbbeddays<- datafile13 %>% 
+  group_by(Healthboard,age_grp,reas1) %>% 
+  summarise(OBDs=sum(OBDs,na.rm=TRUE)) %>% 
+  ungroup()
+
+#add files
+Scotlandhbbeddaysallagesallreasons<-rbind(Scotlandhbbeddays,Scotbeddaysdiffage_grp,hbbeddaysdiffage_grp)
+
+#write_sav(Scotandhbbeddaysallagesallreasons,paste0(filepath,"Scot and hb bed days_R.sav")) # save out file
+
+datafile15<-Scotlandhbbeddaysallagesallreasons %>% mutate(reas1="All")
+
+ScotlandhbAllage_grpsbeddays<- datafile15 %>% 
+  group_by(Healthboard,age_grp,reas1) %>% 
+  summarise(OBDs=sum(OBDs,na.rm=TRUE)) %>% 
+  ungroup()
+
+Scotlandhbbeddaysallagegrpsallreasons<-rbind(ScotlandhbAllage_grpsbeddays,Scotlandhbbeddaysallagesallreasons)
+
+#write_sav(Scotlandhbbeddaysallagegrpsallreasons,paste0(filepath,"Scot and hb bed days all age_grps all reasons.sav")) # save out file
+
+### 15. Bed days LAs ----
+#use la bed days diff age file 
+datafile17<-labeddaysdiffage_grp %>% mutate(age_grp="All")
+table(datafile17$OBDs)
+
+labeddaysallage_grps<- datafile17 %>% 
+  group_by(LocalAuthorityArea,age_grp,reas1) %>% 
+  summarise(OBDs=sum(OBDs,na.rm=TRUE)) %>% 
+  ungroup()
+
+datafile18<-rbind(labeddaysallage_grps,labeddaysdiffage_grp)
+
+#write_sav(datafile18,paste0(filepath,"la bed days.sav")) # save out file
+
+datafile19<-datafile18 %>% mutate(reas1="All")
+
+labeddaysallreason_grp_high_level<- datafile19 %>% 
+  group_by(LocalAuthorityArea,age_grp,reas1) %>% 
+  summarise(OBDs=sum(OBDs,na.rm=TRUE)) %>% 
+  ungroup()
+
+datafile20<-rbind(labeddaysallreason_grp_high_level,datafile18)
+
+#write_sav(datafile20,paste0(filepath,"la bed days all age_grps all reasons.sav")) # save out file
+
+#### 16. Bad Days HB ----
+#match in to hb data sheet template
+hbbedstemplate<-read.csv(paste0(filepath2,"hb_template.csv"))
+
+hbbedstemplate <- hbbedstemplate %>% 
+  rename(Healthboard=hbname,
+         age_grp = age)
+
+
+hbbedstemplate<-hbbedstemplate%>% mutate(Healthboard=toupper(Healthboard),age_grp=toupper(age_grp),reas1=toupper(reas1))
+
+
+#amend variables as necessary in order for matching to work
+
+datafile21 <- left_join(Scotlandhbbeddaysallagegrpsallreasons,hbbedstemplate,
+                        by = c(("Healthboard" = "Healthboard"), ("age_grp"="age_grp"),("reas1"="reas1")))
+
+arrange(datafile21,Healthboard,age_grp,reas1) # arrange data in order for tables
+
+#write_sav(datafile21,paste0(filepath,"hb bed days data sheet minus standard.sav")) # save out file
+datafile21<-datafile21%>% mutate(Healthboard=toupper(Healthboard),age_grp=toupper(age_grp),reas1=toupper(reas1))
```

Comment:
See "Pipes" and "Incremental Data-Frame Names" sections in general feedback.
Why are we changing variable names to be upper case?
```suggestion
datafile %<>% mutate(Healthboard = toupper(Healthboard), age_grp = toupper(age_grp), reas1 = toupper(reas1))
```